src/lib/createGHA/index.ts

```diff
@@ -0,0 +1,234 @@
+      if (arg.defaultOption) return val;
+      // obfuscate the key in a GitHub secret
+      if (arg.name === 'key') return `--key=$\{{ secrets.${GITHUB_SECRET_NAME} }}`;
+      // remove the GitHub flag
+      if (arg.name === 'github') return false;
+      // if a boolean value, return the flag
+      if (arg.type === Boolean && val) return `--${arg.name}`;
+      if (val) return `--${arg.name}=${val}`;
+      return false;
+    })
+    .filter(Boolean)
+    .join(' ');
+
+  return `${command} ${optsString}`.trim();
+}
+
+/**
+ * Function to return various git attributes needed for running GitHub Action
+ */
+export async function getGitData() {
+  // Expressions to search raw output of `git remote show origin`
+  const headRegEx = /^ {2}HEAD branch: /g;
+  const headLineRegEx = /^ {2}HEAD branch:.*/gm;
+
+  const isRepo = await git.checkIsRepo().catch(e => {
+    debug(`error running git repo check: ${e.message}`);
+    return false;
+  });
+
+  let containsGitHubRemote;
+  let containsNonGitHubRemote;
+  let defaultBranch;
+  const rawRemotes = await git.remote([]).catch(e => {
+    debug(`error grabbing git remotes: ${e.message}`);
+    return '';
+  });
+
+  if (rawRemotes) {
+    const remote = rawRemotes.split('\n')[0];
+    const rawRemote = await git.remote(['show', remote]);
+    // Extract head branch from git output
+    const rawHead = headLineRegEx.exec(rawRemote as string)?.[0];
+    if (rawHead) defaultBranch = rawHead.replace(headRegEx, '');
+
+    // Extract the word 'github' from git output
+    const remotesList = (await git.remote(['-v'])) as string;
+    // This is a bit hairy but we want to keep it fairly general here
+    // in case of GitHub Enterprise, etc.
+    containsGitHubRemote = /github/.test(remotesList);
+    containsNonGitHubRemote = /gitlab/.test(remotesList) || /bitbucket/.test(remotesList);
+  }
+
+  const repoRoot = await git.revparse(['--show-toplevel']).catch(e => {
+    debug(`error grabbing git root: ${e.message}`);
+    return '';
+  });
+
+  return { containsGitHubRemote, containsNonGitHubRemote, defaultBranch, isRepo, repoRoot };
+}
+
+/**
+ * Post-command flow for creating a GitHub Action workflow file.
+ *
+ */
+export default async function createGHA(
+  msg: string,
+  command: keyof typeof commands,
+  args: OptionDefinition[],
+  opts: CommandOptions<{}>
+) {
+  const { containsGitHubRemote, containsNonGitHubRemote, defaultBranch, isRepo, repoRoot } = await getGitData();
+
+  if (!opts.github) {
+    if (
+      // not a repo
+      !isRepo ||
+      // in a CI environment
+      isCI() ||
+      // is a repo, but only contains non-GitHub remotes
+      (isRepo && containsNonGitHubRemote && !containsGitHubRemote) ||
+      // not testing this function
+      (process.env.NODE_ENV === 'testing' && !process.env.TEST_CREATEGHA)
+    ) {
+      // We return the original command message and pretend this command flow never happened.
+      return msg;
+    }
+  }
```

Comment:
The tl;dr here is that we should only ask the user (if they're running `rdme` in a non-CI environment) if they want to create a GitHub Action workflow file if they meet any of the following criteria:
- They passed the `--github` flag (duh)
- They have a git repo set up that either has no remotes or a GitHub remote